testgen/main.go

```diff
@@ -38,4 +38,15 @@ func main() {
 	); err != nil {
 		panic(err)
 	}
+
+	err := cbg.Gen{
+		MaxArrayLength:  10,
+		MaxByteLength:   9,
+		MaxStringLength: 8,
+	}.WriteTupleEncodersToFile("testing/cbor_options_gen.go", "testing",
```

Comment:
Any reason not to change this to:

```go
err := cbg.Gen{
	MaxArrayLength:  10,
	MaxByteLength:   9,
	MaxStringLength: 8,
    TupleStructs: []any{
    	types.LimitedStruct{},
    },
    TransparentTypes: ...
}.WriteToFile("testing/ccbor_options_gen.go", "testing")
```

I ask because this would let easily use different representations for different types.

Reply:
no reason other than wanting to keep the scope of this minimal; I was tempted to deprecate the `Write{Type}*` functions and do this kind of thing. Although in my version I was going to go with a `TupleStruct bool` in the config, but moving the types list also into the config could also work.

maybe we should hold off on this for a future change? Because at least with the changes now it's all backward compatible and doesn't require anyone to change anything except if they want different defaults.

Reply:
We could make it backwards compatible, but not cleanly because the function specifies the encoding. My idea was to have separate fields (`TupleStructs`, `TransparentTypes`, `MapStructs`, etc.) for different encodings so we can put them all in the same file.

Alternatively, we can just have multiple files. That's really not an issue, honestly.

Reply:
(I'm not a huge fan of mixing "tuple encoding" and "transparent types" as we do now as they're different encodings)